authorize/log.go

```diff
@@ -161,6 +162,17 @@ func populateLogEvent(
 		return evt.Str(string(field), u.GetEmail())
 	case log.AuthorizeLogFieldHost:
 		return evt.Str(string(field), in.GetAttributes().GetRequest().GetHttp().GetHost())
+	case log.AuthorizeLogFieldIDToken:
+		if s, ok := s.(*session.Session); ok {
+			evt = evt.Str("id-token", s.GetIdToken().GetRaw())
+
+			if t, err := jwt.ParseSigned(s.GetIdToken().GetRaw()); err == nil {
+				var m map[string]any
+				_ = t.UnsafeClaimsWithoutVerification(&m)
+				evt = evt.Interface("id-token-claims", m)
```

Comment:
(no changes required)

Do we already verify the ID token signature somewhere else, and that's why it's fine to fine to call UnsafeClaimsWithoutVerification() here?

Reply:
I don't know about the security ramifications of this. Verifying the signature of the ID token against the configured identity provider would be difficult to implement here (and also likely very slow)

Reply:
It looks like there's some verification related to ID tokens here: https://github.com/pomerium/pomerium/blob/baf8918676aeca3f6f0b017252c1c9560a0d7381/internal/identity/oidc/oidc.go#L225

Do you know if all ID tokens come from that `getIDToken()` method?

Reply:
The ID token comes from the identity provider and is saved to the session.